src/main/java/com/arpnetworking/metrics/common/sources/BaseTcpSource.java

```diff
@@ -84,13 +114,58 @@ public void preStart() {
         @Override
         public Receive createReceive() {
             return receiveBuilder()
-                    .matchEquals(IS_READY, message -> {
-                        getSender().tell(_isReady, getSelf());
+                    .matchEquals(IS_READY, message -> getSender().tell(_isReady, getSelf()))
+                    .matchEquals(UNBIND, unbindRequest -> {
+                        final ActorRef requester = getSender();
+                        if (_connectionActor != null) {
+                            LOGGER.debug()
+                                    .setMessage("Tcp server starting unbinding...")
+                                    .addData("requester", requester)
+                                    .addData("connectionActor", _connectionActor)
+                                    .addData("name", _sink.getName())
+                                    .log();
+                            Patterns.ask(
+                                    _connectionActor,
+                                    TcpMessage.unbind(),
+                                    UNBIND_TIMEOUT.toMillis())
```

Comment:
Unfortunately, the timeout is referenced in two contexts:

* Await.result
* Patterns.ask

In the former it must be a Scala `Duration`, in the latter it can be a `long` or a `Timeout`. Instead of creating both or one from the other at runtime I just used the `Duration` and mapped it into a `long` in the other case.

It's very unfortunate that Akka and Scala don't agree on these types.

Reply:
Please use PatternsCS.ask(... The PatternsCS class uses the Java CompletionStages instread of the Scala Futures

Reply:
Done!